__tests__/app.test.js

```diff
@@ -154,6 +154,36 @@ describe('GET /api/articles', () => {
                 });
             });
     });
+})
+describe('GET /api/articles?topic=', () => {
+    test('200: should return only articles with requested topic', () => {
+        const topicName = "cats";
+        const expectedOutput = data.articleData.filter(({...article}) => article.topic === topicName);
+        return request(app)
+            .get(`/api/articles?topic=${topicName}`)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.articles.length).toBe(expectedOutput.length);
+            });
+    });
+    test('404: should return Not Found if no articles found for valid topic', () => {
+        const topicName = "paper";
+        return request(app)
+            .get(`/api/articles?topic=${topicName}`)
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe('Not Found');
+            });
+    });
+    test('404: should return Not Found if specified topic is not valid', () => {
+        const topicName = "notvalid";
+        return request(app)
+            .get(`/api/articles?topic=${topicName}`)
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe('Not Found');
+            });
+    });
```

Comment:
One missing test here:
We have a valid topic which has no associated articles (paper)
200 : responds with an empty array